encoding/proto/proto.go

```diff
@@ -768,12 +801,35 @@ func decodeNodeEventMessage(pb *internal.NodeEventMessage, m *pilosa.NodeEvent)
 
 func decodeNodeStatus(pb *internal.NodeStatus, m *pilosa.NodeStatus) {
 	m.Node = &pilosa.Node{}
-	decodeNode(pb.Node, m.Node)
-	m.MaxShards = pb.MaxShards.Standard
+	decodeIndexStatuses(pb.Indexes, m.Indexes)
 	m.Schema = &pilosa.Schema{}
 	decodeSchema(pb.Schema, m.Schema)
 }
 
+func decodeIndexStatuses(a []*internal.IndexStatus, m []*pilosa.IndexStatus) {
+	for i := range a {
+		m[i] = &pilosa.IndexStatus{}
```

Comment:
There's an `index out of range` issue here when `a` contains indexes (and `m` does not).
It seems like `m` is always an empty slice, so maybe we just need to set `m` to contain `len(a)` of its zero-value before the for loop. Or do an append in the for loop instead.

You can see the error by starting up a two node cluster, add an index, then restart the cluster; the error will occur after the restart on the second node where `len(a) = 1` and `len(m) = 0`.

Reply:
Good catch. I was hitting a gnarly unit test bug that was caused by this but covered up by a `recover()`. Fixed in f4c9c0f.